src/identity-providers.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+const Identity = require('./identity')
+const OrbitDBIdentityProvider = require('./orbit-db-identity-provider')
+const Keystore = require('orbit-db-keystore')
+const type = 'orbitdb'
+let supportedTypes = {
+  'orbitdb': OrbitDBIdentityProvider,
+}
+
+const getHandlerFor = (type) => {
+  if (!IdentityProviders.isSupported(type)) {
+    throw new Error(`IdentityProvider type '${type}' is not supported`)
+  }
+  return supportedTypes[type]
+}
+
+class IdentityProviders {
+  constructor(options = {}) {
+    this._keystore = options.keystore || Keystore.create(options.keypath || './orbitdb/ipkeys')
+  }
+
+  async sign (identity, data) {
+    const signingKey = await this._keystore.getKey(identity.id)
+    if (!signingKey)
+      throw new Error(`Private signing key not found from Keystore`)
+
+    const signature = await this._keystore.sign(signingKey, data)
+    return signature
+  }
+
+  async verify (signature, publicKey, data) {
+    return this._keystore.verify(signature, publicKey, data)
+  }
+
+  async create(options = {}) {
+    options = Object.assign({}, { type }, options )
+    const IdentityProvider = getHandlerFor(options.type)
+    const identityProvider = new IdentityProvider(options)
+    const id = await identityProvider.createId(options)
+    const { publicKey, idSignature } = await this.signIdentity(id, options)
+    const pubKeyIdSignature = await identityProvider.signIdentity(publicKey + idSignature, options)
+    return new Identity(id, publicKey, idSignature, pubKeyIdSignature, IdentityProvider.type, this)
+  }
+
+  async signIdentity(id, options = {}) {
```

Comment:
It would perhaps make sense to rename the function to something else as we have `signIdentity` in the IdentityProvider. This function doesn't currently sign the Identity, but the `publicKey` for a specific `id`.

Reply:
signPubKeySignature?

Reply:
Woops yea I thought this was the signIdentity in the provider-interface (what you're calling signIdentity)....which proves this whole point. Yea signId is better :+1: and then we can keep signIdentity in the provider